src/gui/sdl_mapper.cpp

```diff
@@ -2101,7 +2096,12 @@ static void SetActiveEvent(CEvent * event) {
 			SetActiveBind(*(mapper.abindit));
 
 			const auto new_bind = dynamic_cast<CKeyBind*>(*mapper.abindit);
-			drop_other_bound_events(new_bind);
+			// TODO: version of drop_other_bound_events that works
+			// for joystick binds, for which above line assigns
+			// nullptr to newbind and causes assertion failure
+			if (new_bind) {
+				drop_other_bound_events(new_bind);
+			}
```

Comment:
I confirmed this fixes #4126 but I'm a little confused on what this `TODO` means.  This seems to work fine as-is with joystick binds.

I would also suggest adding "Regression from f33cc25e57b7d1b8835f1efae7e43efa4ea7691e" to the commit message for future reference since you went through the trouble of bisecting it.  I also confirmed this was the commit that introduced the crash.

Reply:
I updated the first commit to just implement the TODO bits. The problem was that only bindings of host keys were being deduplicated, not bindings of host joystick axes / joystick buttons / joystick hats. The crash was due to a dynamic cast to `CKeyBind*` of something that was a pointer to a different subclass of `CBind` (which returns `nullptr`), then passing that result to the `drop_other_bound_events` function which immediately asserts its argument is not nullptr.

With the updated commit, dedupe is now implemented for all subtypes of CBind, and the downcast is removed.

Reply:
I updated the linux testing procedure to include how I tested the implementation of the TODO bits.